### DISCUSSION TITLE
Can't open links / No context menu / No smooth scrolling

### DISCUSSION THREAD

**Discussion by aa6** [eclipse-theia/theia#13418] [Q&A]
Hello. I'm running
Eclipse Theia IDE
Version 1.46.100
VS Code API Version: 1.85.1
via TheiaIDE.AppImage on Debian 12
and have these problems:

1. I can't open any links. For example when I click `Help -> Documentation` on a Welcome page happens nothing. The same with links on Extensions pages, etc. I'd like to be able to open these links internally and/or with external browser of my choice.

2. When I try to copy selected text from internal pages like Welcome page, I can only do it via Ctrl+C. No context menu appears on a right mouse button click and no other action happening. I've expected a standard context menu with copy button.

3. On internal pages like Welcome page or settings pages, scrolling is not smooth and it causes disorientation because it scrolls too fast and I'm losing it. I'd like to see a smooth scrolling.

So my question is should I create an issue per each of the problems via the bugtracker? Or something of listed is may to be a default behaviour or something else? Thank you.

**Comment by msujew** [comment]
Hey @aa6,

> I can't open any links. For example when I click Help -> Documentation on a Welcome page happens nothing. The same with links on Extensions pages, etc. I'd like to be able to open these links internally and/or with external browser of my choice.

Might be a Linux/Debian issue. It opens the links in my default browser just fine.

> When I try to copy selected text from internal pages like Welcome page, I can only do it via Ctrl+C. No context menu appears on a right mouse button click and no other action happening. I've expected a standard context menu with copy button.

We orient ourselves at VSCode's UI (you might have noticed). It doesn't offer anything either, and a right mouse button click does nothing there as well.

> On internal pages like Welcome page or settings pages, scrolling is not smooth and it causes disorientation because it scrolls too fast and I'm losing it. I'd like to see a smooth scrolling.

Can you show how it looks like? Scrolling is smooth on my side. Might be another Linux specific issue.

Feel free to open issues about these problems though. I haven't specifically looked for existing issues on these, but given that I can't reproduce them, they're likely unique.

**Reply by aa6** [reply]
> Might be a Linux/Debian issue.
> Might be another Linux specific issue.

Highly possible since it's a recently installed IceWM on a bare Debian and I did not any tuning of it. However, both Chromium and LibreWolf browsers installed from a repo do have smooth scrolling by default so I'd expected that from an electron application like Theia too.

> We orient ourselves at VSCode's UI (you might have noticed). It doesn't offer anything either, and a right mouse button click does nothing there as well.

Never used VSCode before and it feels like something is broken. Maybe that policy worth reconsidering regarding that particular problem. But I understand your point.

> Can you show how it looks like?

Sure, here is the MP4 screencast:
https://github.com/eclipse-theia/theia/assets/17766252/3f86aec7-083e-4615-bfa3-a2d861b30d83

> but given that I can't reproduce them, they're likely unique.

I will try to tune the DE first. Thank you for the answers.
